centml/compiler/backend.py

```diff
@@ -134,7 +134,7 @@ def _wait_for_status(self, model_id: str) -> bool:
                 tries += 1
                 self._compile_model(model_id)
             else:
-                tries += 1
```

Comment:
why did we change this?

Reply:
It makes more sense to me that a unexpected status raises an exception rather than simply retrigger compilation.

when I was debugging this inconsistency issue, not raising an exception hid the fact that an incorrect status format was being returned.

Let me know what you think

Reply:
this is not retriggering the compilation right? it is just retrying the status check.

the main reason for having this else condition is for backward compatibility. Even if we have more compilation status added from the server side later on, the client won't fail.

Reply:
yes you're right that this just retrying the status check. 

I didn't consider backwards compatibility, that's a fair point. It does feel like you're kicking the problem down the road if this is your approach to unknown status returns from the server. If the server keeps returning the unknown status, eventually `tries` will exceed the max amount allowed. But I guess it's possible the server starts returning statuses the backend knows how to handle.

This is all pretty minor so I'm fine revert my change on this one